File: Javascript/Two Pointers/E-26. Remove Duplicates from Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let uniqueIndex = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[uniqueIndex] !== nums[i]) {
      uniqueIndex++;
      nums[uniqueIndex] = nums[i];
    }
  }
  return uniqueIndex + 1; // unique items
};

const nums = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4];
console.log(removeDuplicates(nums)); // 5
console.log(nums); // [0,1,2,3,4]

File: Python/Two Pointers/E-26. Remove Duplicates from Sorted Array.py

from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        unique_index = 0
        for i in range(len(nums)):
            if (nums[unique_index] != nums[i]):
                unique_index = unique_index + 1
                nums[unique_index] = nums[i]

        return unique_index + 1
